--- candl-0.6.2/configure.ac.orig	2014-10-05 17:45:18.162160239 +0200
+++ candl-0.6.2/configure.ac	2014-10-05 17:49:15.388817167 +0200
@@ -117,7 +117,7 @@
   if test "x$with_osl_exec_prefix" != "x"; then
     OSL_LDFLAGS="-L$with_osl_exec_prefix/lib"
   fi
-  OSL_LIBS="$with_osl_prefix/lib/libosl.la -losl"
+  OSL_LIBS="-losl"
 esac
 AM_CONDITIONAL(BUNDLED_OSL, test $with_osl = bundled)
 
@@ -211,7 +211,7 @@
   if test "x$with_piplib_exec_prefix" != "x"; then
     PIPLIB_LDFLAGS="-L$with_piplib_exec_prefix/lib"
   fi
-  PIPLIB_LIBS="$with_piplib_prefix/lib/libpiplib$BITS.la -lpiplib$BITS"
+  PIPLIB_LIBS="-lpiplib$BITS"
 esac
 AM_CONDITIONAL(BUNDLED_PIPLIB, test $with_piplib = bundled)
 
