diff -Nur candl-0.6.2/ChangeLog candl-0.6.3/ChangeLog
--- candl-0.6.2/ChangeLog	2012-03-12 17:19:38.000000000 +0100
+++ candl-0.6.3/ChangeLog	2013-02-19 06:27:35.000000000 +0100
@@ -1,3 +1,11 @@
+2013-02-18  Louis-Noel Pouchet  <pouchet@cs.ucla.edu>
+
+	* configure.ac: Bump to candl-0.6.3.
+
+2012-10-26  Louis-Noel Pouchet  <pouchet@cse.ohio-state.edu>
+
+	* source/dependence.c: Fix 2 bugs in scalar privatization.
+
 2012-03-09  Louis-Noel Pouchet  <pouchet@cse.ohio-state.edu>
 
 	* configure.in: Bump to candl-0.6.2.
--- candl-0.6.2/configure.ac.orig	2015-12-02 21:54:57.513138868 +0100
+++ candl-0.6.3/configure.ac	2015-12-02 21:57:50.153131623 +0100
@@ -38,7 +38,7 @@
 AC_PREREQ(2.53)
 
 dnl Fill here the @bug email adress.
-AC_INIT([candl], [0.6.2], [cedric.bastoul@inria.fr,pouchet@cse.ohio-state.edu])
+AC_INIT([candl], [0.6.3], [cedric.bastoul@inria.fr,pouchet@cs.ucla.edu])
 
 AC_CONFIG_SRCDIR([include/candl/candl.h])
 
--- candl-0.6.2/source/dependence.c.orig	2015-12-02 21:54:57.516472201 +0100
+++ candl-0.6.3/source/dependence.c	2015-12-02 22:01:56.763121274 +0100
@@ -1899,14 +1899,15 @@
           pred->next = next;
         free(tmp);
       }
-      pred = tmp;
+      else
+        pred = tmp;
       tmp = next;
 
-      continue;
+    } else {
+      /* Go to the next victim. */
+      pred = tmp;
+      tmp = tmp->next;
     }
-    /* Go to the next victim. */
-    pred = tmp;
-    tmp = tmp->next;
   }
 }
 
@@ -2122,7 +2123,7 @@
                 scop_usr->scalars_privatizable[k] = -1;
             }
 
-            if (nb_priv == priv_buff_size) {
+            if (nb_priv == priv_buff_size - 2) {
               CANDL_realloc(scop_usr->scalars_privatizable, 
                             int*, (priv_buff_size *= 2) * sizeof(int));
               for (k = nb_priv; k < priv_buff_size; ++k)
